File: library/src/main/java/me/proxer/library/api/PagingEndpoint.java
package me.proxer.library.api;

import javax.annotation.Nullable;

/**
 * Interface for all pageable endpoints.
 *
 * @param <T> The type this endpoint returns.
 * @author Ruben Gees
 */
public interface PagingEndpoint<T> extends Endpoint<T> {

    /**
     * Sets the page to request. Must be zero or higher.
     */
    @SuppressWarnings("EmptyMethod")
    PagingEndpoint<T> page(@Nullable Integer page);
}

File: library/src/main/kotlin/me/proxer/library/api/PagingEndpoint.kt
package me.proxer.library.api

/**
 * Interface for all pageable endpoints.
 *
 * @param <T> The type this endpoint returns.
 *
 * @author Ruben Gees
 */
interface PagingEndpoint<T> : Endpoint<T> {

    /**
     * Sets the page to request. Must be zero or higher.
     */
    fun page(page: Int?): PagingEndpoint<T>
}
